packages/http-server-csharp/eng/scripts/emit-scenarios.js

```diff
@@ -0,0 +1,278 @@
+import ora from "ora";
+import pLimit from "p-limit";
+import { basename, dirname, join, resolve } from "path";
+import pc from "picocolors";
+import { fileURLToPath } from "url";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+const tspConfig = join(__dirname, "tspconfig.yaml");
+
+const basePath = join(__dirname, "../..");
+const logDirRoot = join(basePath, "temp", "emit-scenarios-logs");
+const reportFilePath = join(logDirRoot, "report.txt");
+const testScenarioPath = join(basePath, "test", "scenarios");
+const ignoreFilePath = join(testScenarioPath, ".testignore");
+
+const argv = yargs(hideBin(process.argv))
+  .option("input", {
+    alias: "i",
+    type: "string",
+    description: "Input spec folder",
+    default: join(basePath, "..", "http-specs", "specs"),
+  })
+  .option("build", {
+    type: "boolean",
+    describe: "Build the generated projects",
+    default: true,
+  })
+  .option("interactive", {
+    type: "boolean",
+    describe: "Enable interactive mode",
+    default: false,
+  })
+  .help().argv;
+
+const specDir = resolve(argv.input);
+
+// Initialize the port number
+let portNumber = 65000;
+
+// Remove the log directory if it exists.
+async function clearDirectory(dirRoot) {
+  if (await pathExists(dirRoot)) {
+    await rm(dirRoot, { recursive: true, force: true });
+  }
+}
+
+async function copySelectiveFiles(extension, sourceDir, targetDir) {
+  const files = await globby(extension, { cwd: sourceDir });
+  for (const file of files) {
+    const src = join(sourceDir, file);
+    const dest = join(targetDir, file);
+    await copy(src, dest);
+  }
+}
+
+async function compileSpec(file, options) {
+  const relativePath = file;
+  const fullPath = resolve(specDir, relativePath);
+  const { build, interactive } = options;
+  const patchFileDir = join(testScenarioPath, dirname(relativePath));
+  const outputDir = join(testScenarioPath, "generated", dirname(relativePath));
+  const logDir = join(logDirRoot, dirname(relativePath));
+
+  let spinner;
+  if (interactive) {
+    spinner = ora({ text: `Processing: ${relativePath}`, color: "cyan" }).start();
+  }
+
+  try {
+    if (await pathExists(outputDir)) {
+      if (spinner) spinner.text = `Clearing directory: ${outputDir}`;
+      await rm(outputDir, { recursive: true, force: true });
+    }
+    if (spinner) spinner.text = `Creating directory: ${outputDir}`;
+    await mkdir(outputDir, { recursive: true });
+
+    // Increment the port number for each folder
+    portNumber++;
+
+    // Compile the spec and generate server code
+    if (spinner) spinner.text = `Generating csharp server code: ${relativePath}`;
+    await execa("npx", [
+      "tsp",
+      "compile",
+      fullPath,
+      "--emit",
+      resolve(import.meta.dirname, "../.."),
+      "--config",
+      tspConfig,
+      "--output-dir",
+      join(outputDir, "generated"),
```

Comment:
I think we are about to change the emitter to write everything (including mocks) to the output directory, so we should coordinate on that change.

Reply:
Sounds good. Let's chat tomorrow.

Reply:
`generated` here is no longer needed.